src/sagemaker/session.py

```diff
@@ -3554,7 +3554,10 @@ def get_caller_identity_arn(self):
                 user_profile_desc = self.sagemaker_client.describe_user_profile(
                     DomainId=domain_id, UserProfileName=user_profile_name
                 )
-                if user_profile_desc.get("UserSettings") is not None:
+                if (
+                    user_profile_desc.get("UserSettings") is not None
+                    and "ExecutionRole" in user_profile_desc.get("UserSettings").keys()
```

Comment:
Can the value of user_profile_desc["UserSettings"]["ExecutionRole"] be empty or None? if yes need to add check for that as well.

Can change this condition to - `if user_profile_desc.get("UserSettings", {}).get("ExecutionRole"):`

Reply:
Not in this case, because in Line 3558 we explicitly check for the same, if it's None we will proceed to fallback to getting it from the domain as illustrated in the else case

Reply:
Discussed with Mufi and updated accordingly